pkg/middleware/middleware_test.go

```diff
@@ -0,0 +1,94 @@
+package middleware_test
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/codeready-toolchain/registration-service/pkg/server"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAuthMiddleware(t *testing.T) {
+
+	// create a TokenGenerator and a key
+	tokengenerator := testutils.NewTokenManager()
+	kid0 := uuid.NewV4().String()
+	_, err := tokengenerator.AddPrivateKey(kid0)
+	require.NoError(t, err)
+
+	// create some test tokens
+	identity0 := testutils.Identity {
+		ID: uuid.NewV4(),
+		Username: uuid.NewV4().String(),
+	}
+	emailClaim0 := testutils.WithEmailClaim(uuid.NewV4().String() + "@email.tld")
+	tokenValid, err := tokengenerator.GenerateSignedToken(identity0, kid0, emailClaim0)
+	require.NoError(t, err)
+	tokenInvalidNoEmail, err := tokengenerator.GenerateSignedToken(identity0, kid0)
+	require.NoError(t, err)
+	tokenInvalidGarbage := uuid.NewV4().String()
+
+	// start key service
+	keysEndpointURL := tokengenerator.NewKeyServer().URL
+
+	// create server
+	srv, err := server.New("")
+	require.NoError(t, err)	
+
+	// set the key service url in the config
+	os.Setenv(configuration.EnvPrefix+"_"+"AUTH_CLIENT_PUBLIC_KEYS_URL", keysEndpointURL)
+	assert.Equal(t, keysEndpointURL, srv.Config().GetAuthClientPublicKeysURL(), "key url not set correctly")
+	os.Setenv(configuration.EnvPrefix+"_"+"TESTINGMODE", "true")
+	assert.True(t, srv.Config().IsTestingMode(), "testing mode not set correctly")
```

Comment:
I see that this auth setup is being duplicated in different tests. We should really just have one or two lines of code to generate a token with setup the test token generator with the default key manager in place.
Something like `tokengenerator.GenerateSignedToken()` which should use the default keys under the hood which are already available in the default token parser in the test mode.
Otherwise we will have to pass the whole auth setup over all our tests which requires authentication.

Reply:
Yeah, but `tokengenerator.GenerateSignedToken()` is already doing that. The only difference is that I need to give the key explicitly to the instance of `TokenManager` and I think this is good because it makes the test explicitly state which key to use. I guess this might lead to hard to read test code if we use defaults here as the tests are frequently explicitly manipulate the available keys for the `TokenManager`.

Plus, this test is actually different as it is actually fetching the keys. Other tests directly use a ready prepped `TokenParser`. For the middleware, this is different here as it is more or less a blackbox test.